extensions/clusters/gke/gke_cluster_config.go

```diff
@@ -143,10 +143,14 @@ func clusterAddonsBuilder(clusterAddons *ClusterAddons) *management.GKEClusterAd
 }
 
 func cmekConfigBuilder(cmekConfig *CMEKConfig) *management.CMEKConfig {
-	return &management.CMEKConfig{
-		KeyName:  cmekConfig.KeyName,
-		RingName: cmekConfig.RingName,
+	var CMEKConfig *management.CMEKConfig
+	if cmekConfig != nil {
+		CMEKConfig = &management.CMEKConfig{
+			KeyName:  cmekConfig.KeyName,
+			RingName: cmekConfig.RingName,
+		}
 	}
+	return CMEKConfig
```

Comment:
if we're just adding a nil check, do we have to add a new variable, could we do this way with less addition?

```suggestion

	if cmekConfig == nil {
		return nil
	}
	
	return  &management.CMEKConfig{
 			KeyName:  cmekConfig.KeyName,
 			RingName: cmekConfig.RingName,
 	}
	
```

Reply:
This is much cleaner way to do it.

Reply:
Yes, this works. Thanks Ugur!